### DISCUSSION TITLE
[Chapter 06] 다중 or 중첩 if와 switch 를 막기 위한 interface 설계

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/is-my-code-that-strange#48] [Chapter 06 (조건 분기)] [answered]
해당 챕터의 포커스는 다중 or 중첩 if, switch문으로 코드가 작성되고 있다면 인터페이스를 설계 해볼 것을 권유하고 있습니다.

1. 다중 or 중첩 if, switch가 발견될 때, 인터페이스 설계 고려하기 (사후 처리)
2. 발견되기 이전에 인터페이스 설계 고려하기 (예방)

- 저희 학우분들은 어떤 방식을 선호하시는지 궁금합니다.

**Comment by snoopingh** [comment, marked as answer]
선호하는 방식이라면... 이상적으로는 2번이 맞겠지만, 지금의 저로서는 무조건 1번처럼 할 것 같네요 ㅠ.ㅠ
게다가 업무 특성 상 초기에 설계한 인터페이스와 맞지 않는 방향으로 갑자기 정책(?)같은 것이 바뀔 수도 있을 것 같구요
그래도 책에서 배운 내용을 어떻게든 접목 시킬 수 있도록 앞으로 많은 고민과 연습을 해볼 것 같습니다  :)

**Comment by firewoody237** [comment]
2번은 오버엔지니어링이라고 판단하여 1번 방법을 사용합니다.
(확장 및 변경이 확실한 경우 제외)
